### PR TITLE
First round of memory leak fixes in ebpfapi

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Signed-off-by: Alan Jowett <alanjo@microsoft.com>

## Description

First round of memory leak fixes in ebpfapi and unit_tests.

## Testing

CI/CD

## Documentation

No.

### CODE REVIEW

File: libs/api_common/store_helper_internal.cpp [hunk 1]
```diff
@@ -161,6 +161,9 @@ _load_program_data_information(
         }
 
         auto program_type_name_string = ebpf_down_cast_from_wstring(std::wstring(program_type_name));
+        ebpf_free(program_type_name);
```
Comment (dthaler) [inline libs/api_common/store_helper_internal.cpp:164]: To me it would seem to be simpler to just move the free at 272-273 outside the check for success.

**Review by dthaler — APPROVED** [review]

File: libs/api_common/store_helper_internal.cpp (hunk 1 above)
Comment (Alan-Jowett) [reply libs/api_common/store_helper_internal.cpp:164]: I am not happy with the current programming model in use in this code. It's too easy to accidentally leak memory. I think we should switch it to a [RAII](https://en.cppreference.com/w/cpp/language/raii) pattern.

Comment (dthaler) [reply libs/api_common/store_helper_internal.cpp:164]: For C++ files like this one, I completely agree.

Comment (Alan-Jowett) [reply libs/api_common/store_helper_internal.cpp]: For now I will just apply the suggested changes.

**Comment by dthaler** [comment]
@Alan-Jowett check analysis failure

File: libs/api_common/store_helper_internal.cpp [hunk 2]
```diff
@@ -675,5 +679,8 @@ ebpf_store_load_global_helper_information(
             ebpf_free(helper_prototype);
         }
     }
+    if (helper_name) {
```
Comment (dthaler) [inline libs/api_common/store_helper_internal.cpp:682]: remove 'if' wrapper.  ebpf_free can be called with null, like POSIX malloc is.

**Review by dthaler — DISMISSED** [review]

**Review by dv-msft — DISMISSED** [review]

**Review by delaramamiri — APPROVED** [review]

**Review by dthaler — APPROVED** [review]

**Review by saxena-anurag — APPROVED** [review]

**Pull request merged**